exploration/0475-overriding-extending-namespacing.md

```diff
@@ -0,0 +1,240 @@
+});
+```
+
+This might be expressed in MessageFormat as:
+
+```
+Today is {$today :datetime weekday=long year=numeric month=short day=numeric hour=numeric minute=numeric}
+```
+
+The ICU family of libraries provide a shorthand mechanism called _skeletons_ for accessing
+date and time format options without needing a verbose list of options.
+The same message might look like this with a skeleton:
+
+```
+Today is {$today :datetime skeleton=EEEEyMdjm}
+```
+
+Skeletons are not proposed for inclusion in the default registry
+because they are not universally available in all datetime formatting
+libraries.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+- Developers need to add options to the base functions to suit local needs.
+  For example, ICU's skeletons as part of the `:datetime` function
+  Support for this option needs to be specified for local implemented versions.
+
+- Developers want to write a function and access it from messages.
+
+- Developers want to import 3rd party formatting packages and use the package's
+  features from within messages.
+
+- Developers wish to import two or more formatting packages
+  and these might have the same-named functions.
+  For example, there might be both an HTML `p` and TTS `p`
+  function.
+
+- Translators and tools would like a machine-readable way to find out the names
+  and option values for add-on packages.
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+- Developers must be able to write functions that do not later collide with items in the default registry.
+- Developers must be able to write function add-ons that do not later collide with items in the default registry.
+- Users should be able to tell visually when an add-on feature has been used vs. a built-in
+- Users should be able to resolve conflicts between add-on packages that use the same
+  function names without altering add-on packages
+-
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+- A syntactical prefix must not collide with `nmtoken`, to avoid parsing ambiguities with unquoted literals...
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+The actual addition and provisioning of features to an implementation is implementation specific.
+Implementations are not required to read the registry format defined by MFv2
+or use it for any particular purpose.
+
+> For example, a Java implementation might use the `ServiceProvider` interface to load
+> functionality, while a Node application might use `requires`.
+
+If an implementation supports user-installed formatters, selectors, function options,
+or expression annotations, it must also support providing "namespace" prefixes for
+each installed set of functionality.
+
+In this design, each namespace prefix is a short string and is associated with a URL.
```

Comment:
Is there a URL for the `std` namespace?

Reply:
There should be. I can put a placeholder for it.